src/formatter.rs

```diff
@@ -79,6 +79,18 @@ pub fn parse_int(mut buffer: &[u8]) -> Option<i32> {
     }
 }
 
+#[cfg(feature = "hex")]
+const HEX_BYTE_SIZE: usize = 2;
+
+#[cfg(feature = "hex")]
+#[inline]
+pub fn parse_byte_to_hex(byte: u8) -> heapless::String<HEX_BYTE_SIZE> {
+    use heapless::format;
+
+    format!(HEX_BYTE_SIZE;"{:02x}", byte)
```

Comment:
Hmmm, this works of course, but is really bad for code size.

I'd like it much better if you changed it to some manual bit twiddling and ASCII math